server/text_generation_server/models/causal_lm.py

```diff
@@ -573,7 +580,7 @@ def generate_token(
         # slice the attention mask to the correct shape
         attention_mask = batch.attention_mask[:, : -batch.padding_right_offset]
 
-        logits, past = self.forward(
```

Comment:
Do we use the speculative_logits anywhere?

Reply:
Not here, I would need to implement the logic here, but at least the types are correct.

Reply:
Since we're not using the same token choosers as with flash, I would need a lot more code, which should be in a follow-up, no ?

Reply:
Ok!